src/ci/shared.sh

```diff
@@ -15,6 +15,12 @@
 
 # See http://unix.stackexchange.com/questions/82598
 function retry {
+  # Only do retries on CI, locally they shouldn't really be necessary
+  if [ -z "${CI:-}" ]; then
+      "$@"
+      return
+  fi
+
```

Comment:
As far as I can see, the `$CI` environment variable is not exported inside docker (https://github.com/rust-lang/rust/blob/4bd27fc/src/ci/docker/run.sh#L70-L87) so this will disable retries for any docker builds.

That aside, I'm reluctant to give users a different experience to what we work with in CI.